File: codes/src-tgt/java-cpp/HumanEval_16_count_distinct_characters.cpp
### Java version

import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static long countDistinctCharacters(String string) {
        return string.toLowerCase().chars().distinct().count();
    }
    

}


### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
long count_distinct_characters(std::string string) {
    std::set<char> s;
    for (auto c : string) {
        s.insert(tolower(c));
    }
    return s.size();
}

File: test/before_train/trainset_and_testset/cpp-java/HumanEval_16_count_distinct_characters.java
import java.util.*;
import java.lang.reflect.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static long countDistinctCharacters(String string) {
        long count = 0;
        for (char c : string) {
            count += count_distinct_characters(string);
        }
        return count;
    }
    public static void main(String[] args) {
    assert(countDistinctCharacters(("")) == (0l));
    assert(countDistinctCharacters(("abcde")) == (5l));
    assert(countDistinctCharacters(("abcdecadeCADE")) == (5l));
    assert(countDistinctCharacters(("aaaaAAAAaaaa")) == (1l));
    assert(countDistinctCharacters(("Jerry jERRY JeRRRY")) == (5l));
    }

}
